src/alloc_pool.c

```diff
@@ -44,20 +46,142 @@ DEQ_DECLARE(qd_alloc_type_t, qd_alloc_type_list_t);
 #define PATTERN_BACK  0xbabecafe
 
 struct qd_alloc_item_t {
-    DEQ_LINKS(qd_alloc_item_t);
     uint32_t              sequence;
 #ifdef QD_MEMORY_DEBUG
     qd_alloc_type_desc_t *desc;
     uint32_t              header;
 #endif
 };
 
-DEQ_DECLARE(qd_alloc_item_t, qd_alloc_item_list_t);
+//128 has been chosen because many CPUs arch use an
+//adiacent line prefetching optimization that load
+//2*cache line bytes in batch
+#define CHUNK_SIZE 128/sizeof(void*)
 
+struct qd_alloc_chunk_t {
```

Comment:
So assuming sizeof(void*) in a 64 bit machine is 8, 128/8 = 16
qd_alloc_chunk_t will have 
1. 8 bytes for prev
2. 16 items (qd_alloc_item_ts)
3. 8 bytes for next
======================
Total = 128 + 16 = 144 bytes. 
=======================
But Adjacent Cache-Line Prefetch size ie 128 bytes. Looks like 144 bytes will not fit into one Adjacent Cache-Line Prefetch. So, does it make sense to move "items" as the first member of qd_alloc_chunk_t in which case we could use the DEQ for prev and next

Reply:
That's correct,  the thing is that ideally I should have used next/prev to reach the 128 bytes (ie 128 bytes should be the size of the chunk struct), but I have chosen to order the fields according to the LIFO access pattern (prev is accessed only If the first item was the last accessed) and left the adjacent line prefetcher do the dirty work for the other items not accessed yet.

Reply:
define CHUNK_SIZE 128/sizeof(void*) - 2
That way, like you said, one qd_alloc_chunk_t will fit in one  Adjacent Cache-Line Prefetch

Reply:
I'm gonna take a look to https://software.intel.com/sites/default/files/managed/9e/bc/64-ia-32-architectures-optimization-manual.pdf again, to be sure...let me know wdyt